Dockerfile

```diff
@@ -1,71 +1,35 @@
-# Base container that is used for both building and running the app
-FROM registry.fedoraproject.org/fedora-minimal:30 as base
-ARG RUBY_VERSION="2.6"
-ARG NODEJS_VERSION="11"
-
-RUN \
-  echo -e "[nodejs]\nname=nodejs\nstream=${NODEJS_VERSION}\nprofiles=\nstate=enabled\n" > /etc/dnf/modules.d/nodejs.module && \
-  echo -e "[ruby]\nname=ruby\nstream=${RUBY_VERSION}\nprofiles=\nstate=enabled\n" > /etc/dnf/modules.d/ruby.module && \
-  microdnf install mysql-libs mariadb-connector-c postgresql-libs ruby{,gems} rubygem-{rake,bundler} npm nc hostname \
-  # needed for VNC/SPICE websockets
-  python2-numpy && \
-  microdnf clean all
+FROM quay.io/foreman/foreman:builder-base as builder
 
 ARG HOME=/home/foreman
-WORKDIR $HOME
-RUN groupadd -r foreman -f -g 1001 && \
-    useradd -u 1001 -r -g foreman -d $HOME -s /sbin/nologin \
-    -c "Foreman Application User" foreman && \
-    chown -R 1001:1001 $HOME
-
-# Add a script to be executed every time the container starts.
-COPY entrypoint.sh /usr/bin/
-RUN chmod +x /usr/bin/entrypoint.sh
-ENTRYPOINT ["entrypoint.sh"]
-
-# Temp container that download gems/npms and compile assets etc
-FROM base as builder
-ENV RAILS_ENV=production
-ENV FOREMAN_APIPIE_LANGS=en
-ENV BUNDLER_SKIPPED_GROUPS="test development openid libvirt journald"
-
-RUN \
-  microdnf install redhat-rpm-config git \
-    gcc-c++ make bzip2 gettext tar \
-    libxml2-devel libcurl-devel ruby-devel \
-    mysql-devel postgresql-devel libsq3-devel && \
-  microdnf clean all
-
-ENV DATABASE_URL=sqlite3:tmp/bootstrap-db.sql
+ARG RAILS_ENV=production
+ENV RAILS_SERVE_STATIC_FILES=true
+ENV RAILS_LOG_TO_STDOUT true
 
-ARG HOME=/home/foreman
 USER 1001
 WORKDIR $HOME
+
+# Add the sources again and reinstall the the build dependencies in case of caching the builder-base image
 COPY --chown=1001:1001 . ${HOME}/
-# Adding missing gems, for tzdata see https://bugzilla.redhat.com/show_bug.cgi?id=1611117
-RUN echo gem '"rdoc"' > bundler.d/container.rb && echo gem '"tzinfo-data"' >> bundler.d/container.rb
-RUN bundle install --without "${BUNDLER_SKIPPED_GROUPS}" \
-    --binstubs --clean --path vendor --jobs=5 --retry=3 && \
-  rm -rf vendor/ruby/*/cache/*.gem && \
-  find vendor/ruby/*/gems -name "*.c" -delete && \
-  find vendor/ruby/*/gems -name "*.o" -delete
-RUN npm install --no-optional
+RUN \
+  bundle install && \
+  npm install --no-optional
+
 RUN \
   make -C locale all-mo && \
   bundle exec rake assets:clean assets:precompile db:migrate &&  \
-  bundle exec rake db:seed apipie:cache:index && rm tmp/bootstrap-db.sql
+  bundle exec rake db:seed apipie:cache:index && rm tmp/bootstrap-db.sqlite3
 RUN ./node_modules/webpack/bin/webpack.js --config config/webpack.config.js && npm run analyze && rm -rf public/webpack/stats.json
```

Comment:
if you are running it again, I assume you would need to remove the old assets first? otherwise in the image you will have two versions? (one for each time you compiled assets?) probably rm -rf public/webpack/* will work.

Reply:
Note that we are not running in the the builder-base image: the builder-base is just to pre-load the dependencies